### PR TITLE
OC 10 Compatibility 

### PR DESCRIPTION
**Pull request by tomneedham**

### CODE REVIEW

File: appinfo/app.php [hunk 1]
```diff
@@ -20,6 +20,5 @@
 use OCA\OAuth2\AppInfo\Application;
 
 $app = new Application();
-$app->registerSettings();
 $app->getContainer()->query('UserHooks')->register();
 \OC::$server->getJobList()->add('OC\BackgroundJob\Legacy\RegularJob', ['OCA\OAuth2\BackgroundJob\CleanUp', 'run']);
```
Comment (DeepDiver1975) [inline appinfo/app.php]: we can move this to info.xml as well

File: appinfo/info.xml [hunk 2]
```diff
@@ -9,12 +9,16 @@
     <namespace>OAuth2</namespace>
     <category>other</category>
```
Comment (DeepDiver1975) [inline appinfo/info.xml]: should be 'integration'

File: lib/Controller/SettingsController.php [hunk 3]
```diff
@@ -88,10 +97,18 @@ public function __construct($AppName, IRequest $request,
 	 */
 	public function addClient() {
 		if (!isset($_POST['redirect_uri']) || !isset($_POST['name'])) {
-			return new RedirectResponse('../../settings/admin?sectionid=additional#' . $this->appName);
+			return new RedirectResponse(
+				$this->urlGenerator->linkToRouteAbsolute(
+					'settings.SettingsPage.getAdmin',
+					['sectionid' => 'authentication']
+				) . '#oauth2');
```
Comment (joneug) [inline lib/Controller/SettingsController.php]: Isn't it better to inject the app's name as dependency?

File: lib/Controller/SettingsController.php [hunk 4]
```diff
@@ -88,10 +97,18 @@ public function __construct($AppName, IRequest $request,
...
 		}
 		if (!Utilities::isValidUrl($_POST['redirect_uri'])) {
-			return new RedirectResponse('../../settings/admin?sectionid=additional#' . $this->appName);
+			return new RedirectResponse(
+				$this->urlGenerator->linkToRouteAbsolute(
+					'settings.SettingsPage.getAdmin',
+					['sectionid' => 'authentication']
+				) . '#oauth2');
```
Comment (joneug) [inline lib/Controller/SettingsController.php]: Isn't it better to inject the app's name as dependency?

File: lib/Controller/SettingsController.php (hunk 3 above)
Comment (joneug) [inline lib/Controller/SettingsController.php]: (the same in the other occurrences...)

File: appinfo/info.xml [hunk 5]
```diff
@@ -7,14 +7,21 @@
     <author>Project Seminar "sciebo@Learnweb" of the University of Münster</author>
     <version>0.1.0</version>
     <namespace>OAuth2</namespace>
-    <category>other</category>
+    <category>integration</category>
     <dependencies>
-        <owncloud min-version="9.1" max-version="10"/>
+        <owncloud min-version="10" max-version="10"/>
```
Comment (DeepDiver1975) [inline appinfo/info.xml]: @tomneedham 10.0 is required - please adjust

File: lib/Controller/SettingsController.php [hunk 6]
```diff
@@ -61,14 +67,16 @@ class SettingsController extends Controller {
...
 	 */
 	public function __construct($AppName, IRequest $request,
 								ClientMapper $clientMapper,
 								AuthorizationCodeMapper $authorizationCodeMapper,
 								AccessTokenMapper $accessTokenMapper,
 								RefreshTokenMapper $refreshTokenMapper,
 								$UserId,
-								ILogger $logger) {
+								ILogger $logger,
+								IURLGenerator $urlGenerator) {
```
Comment (DeepDiver1975) [inline lib/Controller/SettingsController.php]: unit test are not reflecting this changed - @tomneedham please adopt

File: appinfo/info.xml (hunk 5 above)
Comment (joneug) [inline appinfo/info.xml]: `.travis.yml` has to be changed as well (removing `stable9.1`).

File: lib/Controller/SettingsController.php (hunk 3 above)
Comment (tomneedham) [inline lib/Controller/SettingsController.php]: True, but since the html anchor was hard coded anyway I didn't think it mattered too much :P

**Pull request merged**